atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SerializableTransaction.java

```diff
@@ -859,6 +860,7 @@ private Transaction getReadOnlyTransaction(final long commitTs) {
                 keyValueService,
                 timelockService,
                 lockWatchManager,
+                () -> lockWatchManager.createThrowawayCache(cache.get()),
```

Comment:
So this is kinda janky, but looks like it works: we get another cache at start ts, but it will be created from the updated value store that has processed the commit update

Reply:
This cache isn't really at _any_ timestamp - it is just the latest from the value store (the only thing that passing in the cache does is tell us which type it is).